src/components/modern/LargeTokenValueView.js

```diff
@@ -25,54 +25,71 @@ import { BigNumber } from 'bignumber.js';
 
 // Components
 import { Spacing } from 'components/modern/Layout';
-import Text from 'components/modern/Text';
+import TokenValueView from 'components/modern/TokenValueView';
+import FiatValueView from 'components/modern/FiatValueView';
 
 // Utils
-import { formatTokenValueWithoutSymbol } from 'utils/format';
 import { appFont, spacing } from 'utils/variables';
+import { useThemeColors } from 'utils/themes';
+import { getAssetValueInFiat } from 'utils/rates';
+import { wrapBigNumber } from 'utils/bigNumber';
+
+// Selectors
+import { useFiatCurrency, useChainRates } from 'selectors';
 
 // Types
 import type { TextStyleProp } from 'utils/types/react-native';
-
+import type { Chain } from 'models/Chain';
 
 type Props = {|
   value: ?BigNumber,
+  assetAddress: string,
+  chain: Chain,
   symbol: string,
   style?: TextStyleProp,
 |};
 
 /**
  * Large (& stylized) component to display token value.
  */
-function LargeTokenValueView({ value, symbol, style }: Props) {
-  if (!value) return null;
+function LargeTokenValueView({ value, assetAddress, chain, symbol, style }: Props) {
+  const fiatCurrency = useFiatCurrency();
+  const colors = useThemeColors();
+  const rates = useChainRates(chain);
+  const balanceInFiat = getAssetValueInFiat(value, assetAddress, rates, fiatCurrency);
+  const balanceInFiatNumber = balanceInFiat ? wrapBigNumber(balanceInFiat) : null;
```

Comment:
```suggestion
  const balanceInFiat = getAssetValueInFiat(value, assetAddress, rates, fiatCurrency);
```

Please allow FiatValueView to accept `value` in both `BigNumber` and `number`. The reasoning for that is, while token values need to be precise but all fiat values are actually rounded (i.e. JS `number`) because the rates we use are of JS `number` type.

Reply:
Done